Core/Src/main.c

```diff
@@ -87,16 +96,36 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  const uint8_t adc_tx_buf[ADC_SPI_NUM_BYTES] = {ADC_SPI_TX_START_BYTE, ADC_SPI_TX_CH0_SE_BYTE, 0};
+  uint8_t adc_rx_buf[ADC_SPI_NUM_BYTES] = {0};
+  uint16_t adc_val = 0;
+  uint16_t comparision_val;
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	   /* Pull the PB8 GPIO (CS) pin low */
+	   HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+	   if (HAL_SPI_TransmitReceive(&hspi1, adc_tx_buf, adc_rx_buf, ADC_SPI_NUM_BYTES, ADC_SPI_TIMEOUT) != HAL_OK) {
```

Comment:
So I would suggest putting an else statement right after the if bec lines 117 to 126 are still running if there are an error but they can be skipped right?